### PR TITLE
Login and sign up

### PR DESCRIPTION
**Pull request by flamesoft**
PR stories : 
- https://www.pivotaltracker.com/story/show/121676105 
- https://www.pivotaltracker.com/story/show/121675679

Activities:
- Displays login form
- Displays sign-up form
- Adds tests for login and sign-up
- Note: the omni auth hasn't been implemented yet
- Note: login with different roles is not implemented yet

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,8 +2,15 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
   def current_user
     User.create(username: 'test', role: 'admin')
   end
+
```
Comment (tochman) [inline app/controllers/application_controller.rb]: I think we need to get rid of this hardcoded `current_user` method.

File: app/views/devise/confirmations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (tochman) [inline app/views/devise/confirmations/new.html.erb]: We are using HAML in this project.

**Comment by tochman** [comment]
Some comments:
- Tests looks good (will go over them i detail later) - great increase in coverage!
- Views needs to be written in HAML
- We are using Foundation and I wonder if the views are using the foundation classes?

**Comment by flamesoft** [comment]
Thanks for the comments. I will make the changes tomorrow.

**Comment by flamesoft** [comment]
The views are using the foundation classes because the classes are global. So no special changes need to make. I made only adjustment to the login and the signup views to center the positions of input fields.

**Comment by tochman** [comment]
Okay, I see some classes like `actions` and `fields`, etc...
Another question, have you configured the application layout with flashes? I can´t find that.( to display errors)

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (tochman) [inline app/controllers/application_controller.rb]: Any decision on this?

**Comment by tochman** [comment]
Okay thx. I see no issues with this PR now. Can you go over this @DanielitoAlbinsson @callea2 @diraulo ?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -24,6 +24,11 @@
     it { is_expected.to validate_presence_of(:username) }
     it { is_expected.to validate_uniqueness_of(:username) }
     it { is_expected.to validate_presence_of(:role) }
+    context 'Validate email' do
+      it { should allow_value('user@example.com').for(:email) }
```
Comment (tochman) [inline spec/models/user_spec.rb]: We are using `is_expected.to` or `expect` syntax in our specs. Plz change to that for consistency.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -24,6 +24,11 @@
     it { is_expected.to validate_presence_of(:username) }
     it { is_expected.to validate_uniqueness_of(:username) }
     it { is_expected.to validate_presence_of(:role) }
+    context 'Validate email' do
```
Comment (tochman) [inline spec/models/user_spec.rb:27]: spacing

**Comment by tochman** [comment]
In my opinion this one is ready to merge

**Pull request merged**